plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/geoip/GeoProcessor.java

```diff
@@ -0,0 +1,224 @@
+        geoData.put("continent_name", continent.getName());
+        geoData.put("region_name", subdivision.getName());
+        geoData.put("city_name", city.getName());
+        geoData.put("timezone", location.getTimeZone());
+        geoData.put("latitude", location.getLatitude());
+        geoData.put("longitude", location.getLongitude());
+        if (location.getLatitude() != null && location.getLongitude() != null) {
+            geoData.put("location", new double[]{location.getLongitude(), location.getLatitude()});
+        }
+        return geoData;
+    }
+
+    private Map<String, Object> retrieveCountryGeoData(InetAddress ipAddress) {
+        SecurityManager sm = System.getSecurityManager();
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+        CountryResponse response = AccessController.doPrivileged(new PrivilegedAction<CountryResponse>() {
+            @Override
+            public CountryResponse run() {
+                try {
+                    return dbReader.country(ipAddress);
+                } catch (IOException | GeoIp2Exception e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+
+        Country country = response.getCountry();
+        Continent continent = response.getContinent();
+
+        Map<String, Object> geoData = new HashMap<String, Object>();
+        geoData.put("ip", NetworkAddress.formatAddress(ipAddress));
+        geoData.put("country_iso_code", country.getIsoCode());
+        geoData.put("country_name", country.getName());
+        geoData.put("continent_name", continent.getName());
+        return geoData;
+    }
+
+    public static class Builder implements Processor.Builder {
+
+        private final Path geoIpConfigDirectory;
+        private final DatabaseReaderService databaseReaderService;
+
+        private String ipField;
+        private String databaseFile = "GeoLite2-City.mmdb";
+        private String targetField = "geoip";
+
+        public Builder(Path geoIpConfigDirectory, DatabaseReaderService databaseReaderService) {
+            this.geoIpConfigDirectory = geoIpConfigDirectory;
+            this.databaseReaderService = databaseReaderService;
+        }
+
+        public void setIpField(String ipField) {
+            this.ipField = ipField;
+        }
+
+        public void setDatabaseFile(String dbPath) {
+            this.databaseFile = dbPath;
+        }
+
+        public void setTargetField(String targetField) {
+            this.targetField = targetField;
+        }
+
+        public void fromMap(Map<String, Object> config) {
+            this.ipField = (String) config.get("ip_field");
+
+            String targetField = (String) config.get("target_field");
+            if (targetField != null) {
+                this.targetField = targetField;
+            }
+            String databaseFile = (String) config.get("database_file");
```

Comment:
yes, i think the grok processor should have its own director in the
'config/ingest' folder like geoip?

On Friday, October 30, 2015, Tal Levy <notifications@github.com> wrote:

> In
> plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/geoip/GeoProcessor.java
> <https://github.com/elastic/elasticsearch/pull/14208#discussion_r43502657>
> :
>
> > +        public void setDatabaseFile(String dbPath) {
> > +            this.databaseFile = dbPath;
> > +        }
> > +
> > +        public void setTargetField(String targetField) {
> > +            this.targetField = targetField;
> > +        }
> > +
> > +        public void fromMap(Map<String, Object> config) {
> > +            this.ipField = (String) config.get("ip_field");
> > +
> > +            String targetField = (String) config.get("target_field");
> > +            if (targetField != null) {
> > +                this.targetField = targetField;
> > +            }
> > +            String databaseFile = (String) config.get("database_file");
>
> oh, nevermind. it gets resolved within the config directory. I see. should
> I do the same for grok patterns then?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/14208/files#r43502657>.
>


-- 
Met vriendelijke groet,

Martijn van Groningen